extensions/functions_set.yaml

```diff
@@ -0,0 +1,15 @@
+%YAML 1.2
+---
+scalar_functions:
+  -
+    name: "index_in"
+    description: >
+      Checks the membership of a value in a set of values
+
+      Returns the 0-based index value of some input `T` if it is present,
```

Comment:
Thanks for clarifying the index base :)

I think there are a couple things to resolve in the description:

1. What does `is present` mean? I'll suggest that it means the needle element is equal to (according substrait's `equals` function) an element in the haystack list.
2. What is the behavior when the needle is `NULL` (regardless of whether there's a `NULL` in the `list`). I believe as a consequence of point 1 it'll be `NULL` since `NULL = NULL` returns `NULL`.

Reply:
There's also `NaN` too ... yay?

Reply:
`NaN` -> ... `NULL` ?

Reply:
Gosh, this is really a fraught issue. See https://github.com/cockroachdb/cockroach/issues/18860 for example.

It seems like the decision to make `nan = nan` be `true` is from postgres's decision:

> in order to allow numeric values to be sorted and used in tree-based indexes, PostgreSQL treats NaN values as equal, and greater than all non-NaN values.

Many other incumbent databases also behave this way.

Pros of IEEE semantics:

1. We don't violate expectations of people who aren't steeped in SQL practicalities.

Pros of not-IEEE semantics:

1. NaNs (can) have a well-defined ordering
1. NaNs can be used in indexes

Thoughts?